// @pixi/react/lib/core/reconciler.js
'use strict';

var _package = require('../package.json.js');
var afterActiveInstanceBlur = require('../helpers/afterActiveInstanceBlur.js');
var appendChild = require('../helpers/appendChild.js');
var beforeActiveInstanceBlur = require('../helpers/beforeActiveInstanceBlur.js');
var clearContainer = require('../helpers/clearContainer.js');
var commitUpdate = require('../helpers/commitUpdate.js');
var createInstance = require('../helpers/createInstance.js');
var createReconciler = require('../helpers/createReconciler.js');
var createTextInstance = require('../helpers/createTextInstance.js');
var detachDeletedInstance = require('../helpers/detachDeletedInstance.js');
var finalizeInitialChildren = require('../helpers/finalizeInitialChildren.js');
var getChildHostContext = require('../helpers/getChildHostContext.js');
var getCurrentUpdatePriority = require('../helpers/getCurrentUpdatePriority.js');
var getInstanceFromNode = require('../helpers/getInstanceFromNode.js');
var getInstanceFromScope = require('../helpers/getInstanceFromScope.js');
var getPublicInstance = require('../helpers/getPublicInstance.js');
var getRootHostContext = require('../helpers/getRootHostContext.js');
var hideInstance = require('../helpers/hideInstance.js');
var hideTextInstance = require('../helpers/hideTextInstance.js');
var insertBefore = require('../helpers/insertBefore.js');
var maySuspendCommit = require('../helpers/maySuspendCommit.js');
var preloadInstance = require('../helpers/preloadInstance.js');
var prepareForCommit = require('../helpers/prepareForCommit.js');
var preparePortalMount = require('../helpers/preparePortalMount.js');
var prepareScopeUpdate = require('../helpers/prepareScopeUpdate.js');
var removeChild = require('../helpers/removeChild.js');
var requestPostPaintCallback = require('../helpers/requestPostPaintCallback.js');
var resetAfterCommit = require('../helpers/resetAfterCommit.js');
var resetFormInstance = require('../helpers/resetFormInstance.js');
var resolveEventTimeStamp = require('../helpers/resolveEventTimeStamp.js');
var resolveEventType = require('../helpers/resolveEventType.js');
var resolveUpdatePriority = require('../helpers/resolveUpdatePriority.js');
var setCurrentUpdatePriority = require('../helpers/setCurrentUpdatePriority.js');
var shouldAttemptEagerTransition = require('../helpers/shouldAttemptEagerTransition.js');
var shouldSetTextContent = require('../helpers/shouldSetTextContent.js');
var startSuspendingCommit = require('../helpers/startSuspendingCommit.js');
var suspendInstance = require('../helpers/suspendInstance.js');
var trackSchedulerEvent = require('../helpers/trackSchedulerEvent.js');
var unhideInstance = require('../helpers/unhideInstance.js');
var unhideTextInstance = require('../helpers/unhideTextInstance.js');
var waitForCommitToBeReady = require('../helpers/waitForCommitToBeReady.js');

"use strict";
const reconcilerConfig = {
  isPrimaryRenderer: false,
  noTimeout: -1,
  NotPendingTransition: null,
  supportsHydration: false,
  supportsMutation: true,
  supportsPersistence: false,
  warnsIfNotActing: false,
  afterActiveInstanceBlur: afterActiveInstanceBlur.afterActiveInstanceBlur,
  appendChild: appendChild.appendChild,
  appendChildToContainer: appendChild.appendChild,
  appendInitialChild: appendChild.appendChild,
  beforeActiveInstanceBlur: beforeActiveInstanceBlur.beforeActiveInstanceBlur,
  cancelTimeout: clearTimeout,
  clearContainer: clearContainer.clearContainer,
  commitUpdate: commitUpdate.commitUpdate,
  createInstance: createInstance.createInstance,
  createTextInstance: createTextInstance.createTextInstance,
  hideTextInstance: hideTextInstance.hideTextInstance,
  unhideTextInstance: unhideTextInstance.unhideTextInstance,
  detachDeletedInstance: detachDeletedInstance.detachDeletedInstance,
  finalizeInitialChildren: finalizeInitialChildren.finalizeInitialChildren,
  getChildHostContext: getChildHostContext.getChildHostContext,
  getCurrentUpdatePriority: getCurrentUpdatePriority.getCurrentUpdatePriority,
  getInstanceFromNode: getInstanceFromNode.getInstanceFromNode,
  getInstanceFromScope: getInstanceFromScope.getInstanceFromScope,
  getPublicInstance: getPublicInstance.getPublicInstance,
  getRootHostContext: getRootHostContext.getRootHostContext,
  hideInstance: hideInstance.hideInstance,
  insertBefore: insertBefore.insertBefore,
  insertInContainerBefore: insertBefore.insertBefore,
  maySuspendCommit: maySuspendCommit.maySuspendCommit,
  preloadInstance: preloadInstance.preloadInstance,
  prepareForCommit: prepareForCommit.prepareForCommit,
  preparePortalMount: preparePortalMount.preparePortalMount,
  prepareScopeUpdate: prepareScopeUpdate.prepareScopeUpdate,
  removeChild: removeChild.removeChild,
  removeChildFromContainer: removeChild.removeChild,
  requestPostPaintCallback: requestPostPaintCallback.requestPostPaintCallback,
  resetAfterCommit: resetAfterCommit.resetAfterCommit,
  resetFormInstance: resetFormInstance.resetFormInstance,
  resolveEventTimeStamp: resolveEventTimeStamp.resolveEventTimeStamp,
  resolveEventType: resolveEventType.resolveEventType,
  resolveUpdatePriority: resolveUpdatePriority.resolveUpdatePriority,
  scheduleTimeout: setTimeout,
  shouldAttemptEagerTransition: shouldAttemptEagerTransition.shouldAttemptEagerTransition,
  setCurrentUpdatePriority: setCurrentUpdatePriority.setCurrentUpdatePriority,
  shouldSetTextContent: shouldSetTextContent.shouldSetTextContent,
  startSuspendingCommit: startSuspendingCommit.startSuspendingCommit,
  suspendInstance: suspendInstance.suspendInstance,
  trackSchedulerEvent: trackSchedulerEvent.trackSchedulerEvent,
  unhideInstance: unhideInstance.unhideInstance,
  waitForCommitToBeReady: waitForCommitToBeReady.waitForCommitToBeReady,
  rendererPackageName: "@pixi/react",
  rendererVersion: _package.default.version
};
const reconciler = createReconciler.createReconciler(reconcilerConfig);
reconciler.injectIntoDevTools();

exports.reconciler = reconciler;
